### PR TITLE
feat(ui): add range to DatePicker

### PR DESCRIPTION
**Pull request by lobunto**
OMEGA-112

## What is the reason for this change?

PR 1 of 2 for OMEGA-112 "add date range for graph".
The DatePicker component should later be moved out to a separate package. It should probably also be used as a hook.

### What is the expected result?

By providing a property `showRange` you can choose a range between two dates.
The result returned from the change event is an object containing `from` and `to` date.

## How is it solved?

Extend the DatePicker component to also return an object with dates instead of just a string.
Add separate styles config for normal DatePicker and DatePicker with range.

## Any changes/updates that may affect others?

|                       | Changed |
| --------------------- | ------- |
| Unit tests            |       |
| Dependencies          |         |
| Environment variables |         |
| Build scripts         |         |

## How can it be tested and verified?

- [ ] You should be able to choose a range between two dates
- [ ] Both selected dates should appear in the input box
- [ ] The design should be according to the design sketches in [figma](https://www.figma.com/file/sJHoScmQRq0hlr6AcHI9WB/Elvia.no?node-id=10567%3A72840) (design QA has been done before the PR was created).

### CODE REVIEW

File: packages/ui/src/components/Datepicker/Datepicker.tsx [hunk 1]
```diff
@@ -158,16 +180,59 @@ export const Datepicker: React.FC<DatepickerProps> = props => {
...
+          to: rangeState.to || rangeState.from,
+        },
+      ]
+    } else {
+      if (isValidDate) {
+        return dayjs(value, format).toDate()
+      } else {
+        return null
+      }
+    }
```
Comment (t0ff3r) [inline packages/ui/src/components/Datepicker/Datepicker.tsx:233]: Should be possible to flatten these nested if/else to simplify a bit?

File: packages/ui/src/components/Datepicker/Datepicker.tsx [hunk 2]
```diff
@@ -158,16 +180,59 @@ export const Datepicker: React.FC<DatepickerProps> = props => {
...
+      callOnRangeChange(range)
+      setRangeState(range)
+    }
+
+    if (!showRange && !(modifiers.disabled || modifiers.outside)) {
+      callOnChange(dayjs(day).format(format))
+
+      setTimeout(() => {
+        setShowDatepicker(false)
+      }, 100)
```
Comment (t0ff3r) [inline packages/ui/src/components/Datepicker/Datepicker.tsx:214]: Is the timeout necessary?

**Review by t0ff3r — APPROVED** [review]
Code looks good except for a few minor comments that are not show stoppers but in general I find this component a too big and complex. We should break out some hooks and clean this one up a bit. Not saying you need to fix it right now. Good work!

**Comment by lobunto** [comment]
> Code looks good except for a few minor comments that are not show stoppers but in general I find this component a too big and complex. We should break out some hooks and clean this one up a bit. Not saying you need to fix it right now. Good work!

Good! Will make a package of it in a separate PR.

File: packages/ui/src/components/Datepicker/Datepicker.tsx (hunk 2 above)
Comment (lobunto) [reply packages/ui/src/components/Datepicker/Datepicker.tsx:214]: I cannot see what function it does and there is no difference by removing it.

Comment (lobunto) [reply packages/ui/src/components/Datepicker/Datepicker.tsx:214]: Well, the datepicker is actually a bit laggy without the timeout when clicking on a date. So added it back for now.

**Pull request merged**